lib/Ride.dart

```diff
@@ -260,3 +270,224 @@ class Ride {
+                          Column(
+                            crossAxisAlignment: CrossAxisAlignment.start,
+                            children: <Widget>[
+                              Text('Driver', style: TextStyle(fontSize: 11)),
+                              Text(ride.type == 'active' ? 'Confirmed' : 'TBD', style: CarriageTheme.rideInfoStyle)
+                            ],
+                          )
+                        ],
+                      ) : Container()
+                    ]
+                ),
+              ),
+              showArrow ? Icon(Icons.chevron_right, size: 28) : Container()
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class RecurringRidesGenerator {
+  RecurringRidesGenerator(this.originalRides);
+  List<Ride> originalRides;
+
+  int daysUntilWeekday(DateTime start, int weekday) {
+    int startWeekday = start.weekday;
+    if (weekday < startWeekday) {
+      weekday += 7;
+    }
+    return weekday - startWeekday;
+  }
+
+  DateTime nextDateOnWeekday(DateTime start, int weekday) {
+    int daysUntil = daysUntilWeekday(start, weekday);
+    return start.add(Duration(days: daysUntil));
+  }
+
+  bool ridesEqualTimeLoc(Ride ride, Ride otherRide) {
+    return (
+        ride.startTime.isAtSameMomentAs(otherRide.startTime)
+            && ride.endTime.isAtSameMomentAs(otherRide.endTime)
+            && ride.startLocation == otherRide.startLocation
+            && ride.endLocation == otherRide.endLocation
+            && ride.startAddress == otherRide.startAddress
+            && ride.endAddress == otherRide.endAddress
+    );
+  }
+
+  bool wasDeleted(Ride generatedRide, List<Ride> deletedRides) {
+    return deletedRides.where((deletedRide) => ridesEqualTimeLoc(deletedRide, generatedRide)).isNotEmpty;
+  }
+
+  List<Ride> generateRideInstances() {
+    List<Ride> allRides = [];
+    Map<String, Ride> originalRidesByID = Map();
+    Map<Ride, String> rideInstanceParentIDs = Map();
+
+    for (Ride originalRide in originalRides) {
+      originalRidesByID[originalRide.id] = originalRide;
+    }
+    for (Ride originalRide in originalRides) {
+      // add one time rides
+      if (!originalRide.deleted && !originalRide.recurring) {
+        allRides.add(originalRide);
+      }
+
+      if (originalRide.recurring) {
+        Duration rideDuration = originalRide.endTime.difference(originalRide.startTime);
+        List<Ride> deletedInstances = originalRide.edits.map((rideID) => originalRidesByID[rideID]).where((ride) => ride.deleted).toList();
+        List<int> days = originalRide.recurringDays;
+
+        // find first occurrence
+        int daysUntilFirstOccurrence = days.map((day) => daysUntilWeekday(originalRide.startTime, day)).reduce(min);
+        DateTime rideStart = originalRide.startTime.add(Duration(days: daysUntilFirstOccurrence));
+        int dayIndex = days.indexOf(rideStart.weekday);
+
+        // generate instances of recurring rides
```

Comment:
The comments throughout all this code really helped explain your implementation :)